crates/timers/src/lib.rs

```diff
@@ -67,10 +67,22 @@ TODO
 #[cfg(feature = "futures")]
 extern crate futures_rs as futures;
 
+use js_sys::Function;
 use wasm_bindgen::prelude::*;
 
-fn window() -> web_sys::Window {
-    web_sys::window().unwrap_throw()
+#[wasm_bindgen]
+extern "C" {
+    #[wasm_bindgen(js_name = "setTimeout")]
+    pub fn set_timeout(handler: &Function, timeout: i32) -> i32;
```

Comment:
Rather than taking `&js_sys::Function`, if we're going to write the bindings by-hand, we should make them take `&Closure<FnMut()>`

Also, let's put these raw bindings into a `sys` module.

Reply:
I moved the bindings to a `sys` module, but I'm not sure about changing the signature to accept a Closure. The problem is the `callback` module is using `js_sys::Promise` and passing the `resolve: &Function` parameter straight into the set_timeout. It was already doing that before, since the main web_sys api for [`set_timeout_with_callback_and_arguments`](https://rustwasm.github.io/wasm-bindgen/api/web_sys/struct.WorkerGlobalScope.html#method.set_timeout_with_callback_and_timeout_and_arguments) accepts a `Function`.

Reply:
Ah, this is a good point; I forgot that it was using promises like that.